app/models/authorization.rb

```diff
@@ -0,0 +1,108 @@
+require 'bcrypt'
+# class Authorization < ActiveRecord::Base
+class Authorization < OmniAuth::Identity::Models::ActiveRecord
+  belongs_to :customer, foreign_key: "customer_id"
+  validates :uid,
+            uniqueness: true,
+            format: {:with => /\A\S+@\S+\z/}, 
+            allow_blank: true,
+            on: :create
+  before_validation do
+    if self.password_digest.blank? && self.password.blank?
```

Comment:
I see, this is for customer without a password and created by staff. But do they need to go through authentication in that way(cause they will only be accessed by staff)?

Reply:
That's why they are skipping validation. There isn't a good way to skip it other than this that I could think of (without adding more db columns or something)